### PR TITLE
 feat: Replace multiple SNMP walks with one SNMP Bulk 

### PR DESCRIPTION
**Pull request by omrozowicz-splunk**

### CODE REVIEW

**Comment by omrozowicz-splunk** [comment]
I have read the Code of Conduct and I hereby accept the Terms

File: splunk_connect_for_snmp_poller/manager/tasks.py [hunk 1]
```diff
@@ -72,98 +147,23 @@ def snmp_polling(
...
-                                    )
-                            except Exception as e:
-                                logger.error(
-                                    f"Invalid format for oid. Error message: {e}"
-                                )
+                # Divide varBinds for WALK/BULK actions
+                varbind_collection = sort_varbinds(varBinds)
+                logger.info(f"Varbind collection: {varbind_collection}")
+                # Perform SNMP BULK
+                get_data(varbind_collection.bulk, bulk_handler, *static_parameters)
```
Comment (lstoppa) [inline splunk_connect_for_snmp_poller/manager/tasks.py:154]: Change this name please. Maybe get_snmp_data? get_data looks like it's performing only an SNMP GET data

File: splunk_connect_for_snmp_poller/manager/tasks.py [hunk 2]
```diff
@@ -72,98 +147,23 @@ def snmp_polling(
...
-                    host,
-                    port,
-                    profile,
-                    mib_server_url,
-                    index,
-                    otel_logs_url,
-                    otel_metrics_url,
-                    one_time_flag,
-                )
+                walk_handler(profile, *static_parameters)
```
Comment (lstoppa) [inline splunk_connect_for_snmp_poller/manager/tasks.py:162]: I'd rename walk_handler in snmp_walk_handler, and get_handler in snmp_get_handler

**Review by lstoppa — APPROVED** [review]
Looks good!

File: splunk_connect_for_snmp_poller/manager/tasks.py (hunk 1 above)
Comment (omrozowicz-splunk) [reply splunk_connect_for_snmp_poller/manager/tasks.py]: Done

File: splunk_connect_for_snmp_poller/manager/task_utilities.py [hunk 3]
```diff
@@ -126,104 +133,67 @@ def get_translated_string(mib_server_url, varBinds):
...
-):
+def mib_string_handler(mib_list: list) -> VarbindCollection:
     """
     Perform the SNMP Get for mib-name/string, where mib string is a list
     1) case 1: with mib index - consider it as a single oid -> snmpget
     e.g. ['SNMPv2-MIB', 'sysUpTime',0] (syntax -> [<mib_file_name>, <mib_name/string>, <min_index>])
-    execute snmpget to query info correlated to this specific mib-name/string
 
-    2) case 2: without mib index - consider it as a oid with * -> snmpwalk
+    2) case 2: without mib index - consider it as a oid with * -> snmpbulk
```
Comment (weliasz) [inline splunk_connect_for_snmp_poller/manager/task_utilities.py:142]: I think this is still snmp walk but implemented using getbulk not get command.
Actually this is equivalent of shell command snmpbulkwalk.

**Review by weliasz — APPROVED** [review]

File: splunk_connect_for_snmp_poller/manager/task_utilities.py (hunk 3 above)
Comment (omrozowicz-splunk) [reply splunk_connect_for_snmp_poller/manager/task_utilities.py]: Changed it.

**Comment by omrozowicz-splunk** [comment]
I have read the CLA Document and I hereby sign the CLA

**Pull request merged**